src/renderer/components/ProjectCreation/ModalContainer.tsx

```diff
@@ -57,11 +61,28 @@ const ModalContainer = ({ isOpen, closeModal }: Props) => {
     }
   };
 
-  const viewComponents = [
-    NewProjectView,
-    ImportMediaView,
-    RunTranscriptionView,
-  ];
+  const viewComponents: any = useMemo(() => {
+    return [NewProjectView, ImportMediaView, RunTranscriptionView];
+  }, []);
+
+  useEffect(() => {
+    const fetchIfCloudConfigRequired = async () => {
+      const result = await requireCloudConfig();
+      return result;
+    };
+
+    fetchIfCloudConfigRequired()
+      .then((isConfigRequired) => {
+        if (isConfigRequired) {
+          viewComponents.splice(1, 0, CloudConfigView);
+        }
+        // for linter
+        return isConfigRequired;
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, [viewComponents]);
```

Comment:
@chloebrett Could you go over this?

It adds the `CloudConfigView` to the project creation modals when necessary. But when you create a project then go back to the home page (after creation and saving)... the `CloudConfigView` gets re-inserted to the array (or most likely is just not re-rendering thus the array is unchanged).

If you close the app completely and re-open it (with the `cloudConfig.txt` in `~/.config/Electron/mlvet/`) it wont add the `CloudConfigView` to the array...

How can we make it such that if the user sets an API key then creates a project then heads back to the home page and creates a second project it doesnt go through the cloud view again

Reply:
I would do this

1. Make a state variable `isCloudConfigRequired`
2. In your useEffect, set that variable to true or false when the component loads based on the output of the API call
3. In the `useMemo` that builds the viewComponents, make it add the `CloudCOnfigView` if `isCloudConfigRequired` is true, and otherwise it just returns the other three

Just don't mutate variables directly, react won't know to re-render if you do